src/Plugin/Block/DisplayCitationsBlock.php

```diff
@@ -176,9 +176,11 @@ public function blockSubmit($form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function getCacheTags() {
-    // Retrieve the node ID.
+    // Retrieve the node parameter.
     $node = $this->routeMatch->getParameter('node');
-    $node_id = $node instanceof NodeInterface ? $node->id() : NULL;
+
+    // If it's an object, get the ID. Otherwise, assume it's already the ID.
+    $node_id = is_object($node) ? $node->id() : $node;
```

Comment:
this isn't needed because $node instanceof NodeInterface is already being checked? The module version just needs to be upgraded in the environment